src/taskgraph/files_changed.py

```diff
@@ -8,23 +8,45 @@
 
 
 import logging
+import os
 
 import requests
 from redo import retry
 
 from .util.memoize import memoize
 from .util.path import match as match_path
+from .util.vcs import get_repository
 
 logger = logging.getLogger(__name__)
 
 
 @memoize
-def get_changed_files(repository, revision):
+def get_changed_files(head_repository_url, head_rev, base_rev=None):
     """
-    Get the set of files changed in the push headed by the given revision.
+    Get the set of files changed between revisions.
     Responses are cached, so multiple calls with the same arguments are OK.
     """
-    url = "{}/json-automationrelevance/{}".format(repository.rstrip("/"), revision)
+    repo_path = os.getcwd()
+    repository = get_repository(repo_path)
+
+    if repository.tool == "hg":
+        # TODO Use VCS version once tested enough
+        return _get_changed_files_json_automationrelevance(
+            head_repository_url, head_rev
+        )
+    elif repository.tool == "git":
+        return repository.get_changed_files(rev=head_rev, base_rev=base_rev)
+
+    raise ValueError(f"Unknown repo tool: {repository.tool}")
```

Comment:
Might as well change the `elif` to and `else` and get rid of this `raise ValueError`. I believe `get_repository` will already raise if the repo is not either `hg` or `git`.

Reply:
👌 done!